warehouse/models/staging/payments/elavon/stg_elavon__transactions.sql

```diff
@@ -1,108 +1,96 @@
+        {{ trim_make_empty_string_null('roc_text') }} AS roc_text,
+        {{ trim_make_empty_string_null('trn_aci') }} AS trn_aci,
+        {{ trim_make_empty_string_null('card_scheme_ref') }} AS card_scheme_ref,
+        {{ trim_make_empty_string_null('trn_ref_num') }} AS trn_ref_num,
+        {{ trim_make_empty_string_null('settlement_method') }} AS settlement_method,
+        {{ trim_make_empty_string_null('currency_code') }} AS currency_code,
+        {{ trim_make_empty_string_null('cb_acq_ref_id') }} AS cb_acq_ref_id,
+        {{ trim_make_empty_string_null('chgbk_rsn_code') }} AS chgbk_rsn_code,
+        {{ trim_make_empty_string_null('chgbk_rsn_desc') }} AS chgbk_rsn_desc,
+        {{ trim_make_empty_string_null('mer_ref') }} AS mer_ref,
+        {{ trim_make_empty_string_null('purch_id') }} AS purch_id,
+        {{ trim_make_empty_string_null('cust_cod') }} AS cust_cod,
+        {{ trim_make_empty_string_null('trn_arn') }} AS trn_arn,
+        {{ trim_make_empty_string_null('term_id') }} AS term_id,
+        {{ trim_make_empty_string_null('ent_num') }} AS ent_num,
+        dt,
+        execution_ts
+
+    FROM source
+
 ),
 
-batch_b_transactions AS (
+format_dates AS (
+
     SELECT
-      transactions.payment_reference,
-      transactions.payment_date,
-      transactions.account_number,
-      transactions.routing_number,
-      transactions.fund_amt,
-      transactions.batch_reference,
-      transactions.batch_type,
-      transactions.customer_batch_reference,
-      transactions.customer_name,
-      transactions.merchant_number,
-      transactions.external_mid,
-      transactions.store_number,
-      transactions.chain,
-      transactions.batch_amt,
-      transactions.amount,
-      transactions.surchg_amount,
-      transactions.convnce_amt,
-      transactions.card_type,
-      transactions.charge_type,
-      transactions.charge_type_description,
-      transactions.card_plan,
-      transactions.card_no,
-      transactions.chk_num,
-      transactions.transaction_date,
-      transactions.settlement_date,
-      transactions.authorization_code,
-      transactions.chargeback_control_no,
-      transactions.roc_text,
-      transactions.trn_aci,
-      transactions.card_scheme_ref,
-      transactions.trn_ref_num,
-      transactions.settlement_method,
-      transactions.currency_code,
-      transactions.cb_acq_ref_id,
-      transactions.chgbk_rsn_code,
-      transactions.chgbk_rsn_desc,
-      transactions.mer_ref,
-      transactions.purch_id,
-      transactions.cust_cod,
-      transactions.trn_arn,
-      transactions.term_id,
-      transactions.ent_num,
-      transactions.dt,
-      transactions.execution_ts
-    FROM {{ source('elavon_external_tables', 'transactions') }}
-    WHERE batch_type = 'B'
-    AND execution_ts = (SELECT MAX(execution_ts) FROM {{ source('elavon_external_tables', 'transactions') }})
+
+        * EXCEPT (payment_date, transaction_date, settlement_date),
+
+        CASE WHEN
```

Comment:
When you have three identical blocks like this, I would actually just make a macro and use that to reduce risk of someone editing and missing one. So like I'd make a macro called `parse_elavon_date` that has this case logic and takes column name as a variable and call that on these.